### DISCUSSION TITLE
New layouts in Freeplane Preview version 1.11.1_01 alpha

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#819] [Announcements]
Hello folks,

in the latest preview, you can try the new layout options shown in the attached screenshot.

<img width="594" alt="layoutExamples" src="https://user-images.githubusercontent.com/1558052/201489737-9aa96077-6f29-42d8-ade4-3295a3e827df.png">

Generally, the variants containing a gear symbol on the buttons are smarter because the layout may depend on the node position and the ancestor node layouts.

The example mind map shown there is here:

[layoutExamples.mm.zip](https://github.com/freeplane/freeplane/files/9995688/layoutExamples.mm.zip)

The preview version can be downloaded from https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

Try it out and leave your suggestions about the logic and the UI.

Best, Dimitry

**Comment by guoranjun2** [comment]
Thank you for your great job, Dimitry. I have been waiting too long for this.

**Comment by guoranjun2** [comment]
Is it possible to provide layout of timeline in later versions? sometimes, It's a more proper way to display information in the limited screen. 

![image](https://user-images.githubusercontent.com/50692082/201501765-1d3bb196-b6a6-4599-ba7d-c57def901d27.png)

**Reply by dpolivaev** [reply]
If two nodes are connected by an edge, one of them is a parent and another one is a child because edges connect only parents with their child nodes.

**Comment by reregaga** [comment]
Clouds extend beyond the parent cloud, it doesn't look pretty, I would like the parent cloud to automatically increase in size.
>![cloud](https://user-images.githubusercontent.com/76629445/201548814-b2a8d2c5-9cd6-4b5c-9148-90018441d8cd.png)

Also, I [still](https://github.com/freeplane/freeplane/issues/795) can't save mindmap in `XML (.xml)` format, although other formats like `Export raw xml with node formatting (.xml)` and `Microsoft Project 2003 XML (.xml)` save fine.
>![xml](https://user-images.githubusercontent.com/76629445/201548884-1a487240-a1d7-45ab-a806-35cd4300d245.png)

>![error](https://user-images.githubusercontent.com/76629445/201548907-f27942b1-de97-4921-a634-dde3601b9332.png)

I also think that we should rename `opml file (.opml)` to `OPML file (.opml)` and `Export raw xml with node formatting (.xml)`  to `Export raw XML with node formatting (.xml)` for more consistency.

**Reply by dpolivaev** [reply]
I am sorry, but this thread is not related to OPML and XML topics.
About the clouds, a bug report already has been submitted.

**Comment by euu2021** [comment]
👏Nice!! I will start testing. 

Some things that I noticed at first use:
-1: Looking at the preview of this layout:
![image](https://user-images.githubusercontent.com/77707706/201957675-ec45885f-8c08-4a22-9ce4-87eb95a44678.png)

I would expect that the result would be distributed like this layout:
![image](https://user-images.githubusercontent.com/77707706/201957948-b86f8f64-7c71-47ea-b7d1-bcea72efcbcb.png)

Because, in the preview, the nodes are distributed to the left and right:
![image](https://user-images.githubusercontent.com/77707706/201960649-eb72eb45-a874-4941-afb5-26fe9bd259d5.png)

Probably, my confusion is because I have no idea of the difference between the layouts with and without the gear.

-2: to me, one of the strongest points of FP is automatic layout that avoids overlapping, because it allows great scalability to the growth of the map. So, I think it would be useful if the layouts avoided overlapping. For example, right now, the layout to the left results in this overlapping:
![image](https://user-images.githubusercontent.com/77707706/201959889-80b676d0-c9be-4cd6-bb69-9a4651337118.png)
And what I suggest would result in this (I manually positioned the bbbb node):
![image](https://user-images.githubusercontent.com/77707706/201960227-ca0ab868-6d83-453b-9d6c-744290c84976.png)

**Reply by dpolivaev** [reply]
Generally, placing two questions in one post makes the answering harder when the questions are unrelated.

**Reply by dpolivaev** [reply]
Layouts with the gear are meant so that they somehow depend on node position (left/right or top/bottom) and the ancestor node layouts. Layouts without the gear have the same meaning at any node position.

Generally my work on layouts is still work in progress, so I am going to introduce many further options. At the end I am going to ask which to keep, which to remove and which to change.

**Reply by dpolivaev** [reply]
Overlapping is unavoidable in some cases because Freeplane tries to keep the position of the node relative to the root node when it is folded or unfolded. Here Freeplane behavior is different from e.g. SimpleMind. It gives the user a smoother experience because nodes are not suddenly moved.

Disabling such variants would restrict the number of available layouts. Therefore they are supported although users should use them only in cases where they can position the nodes so that the overlap does not happen.

All layouts shown in the current preview versions are drafts, and we can discuss them after I finish implementing the ideas currently in my mind.

**Reply by euu2021** [reply]
I see, thanks. 

I think the overlapping problem is an interesting point to be discussed. I wonder what is the most useful for the general user: not overlapping X not having nodes jumping around when folding/unfolding.

Just to add examples: Mindmanager also adopts measures to avoid overlapping, as shown in the examples at #501. EDIT: actually, I think those examples are not valid, because I don't know if the user placed the nodes manually.

**Reply by dpolivaev** [reply]
Freeplane does not create overlapping nodes unless you reverse the child node direction. And it is up to you.
Reversing direction can be useful because it allows you to have nodes at both sides of their parent node.
And only users decide which layout combinations are useful for them and which are not.

**Comment by dpolivaev** [comment]
There is a new preview version 1.11.1_03 alpha with more compact layouts following suggestions from @cesar404 in https://github.com/freeplane/freeplane/discussions/825. 
Unfortunately, it is not fully compatible with the previous 1.11.1 previews.

Please check it out and share your feedback.

**Comment by Illegit** [comment]
Hi the preview with java has been corrupted, please reload it.

**Comment by Illegit** [comment]
Thanks

**Comment by cqg2** [comment]
Thanks Dimitry, this new feature is really great and works really well!

**Comment by sgjohnson1981** [comment]
These are neat features I'd like to try. But the installers, including the preview installers, don't allow for changing the install directory, at least not that I can find. I didn't really want to install over my stable version. Is the portable version the only workaround? Does it include Java?

**Reply by sgjohnson1981** [reply]
Nevermind. I figured out how to change the java home variable in the config file to a java install I just threw into the folder.

**Comment by Bommi46** [comment]
thank you sooooo much for the new layout-options!
